crates/bevy_app/src/main_schedule.rs

```diff
@@ -138,15 +138,19 @@ impl MainScheduleOrder {
 }
 
 impl Main {
-    /// A system that runs the "main schedule"
-    pub fn run_main(world: &mut World, mut run_at_least_once: Local<bool>) {
-        if !*run_at_least_once {
-            let _ = world.try_run_schedule(PreStartup);
-            let _ = world.try_run_schedule(Startup);
-            let _ = world.try_run_schedule(PostStartup);
-            *run_at_least_once = true;
-        }
+    /// Runs the startup phase.
+    ///
+    /// Must be called once before the main schedule is started.
+    pub fn startup(world: &mut World) {
+        let _ = world.try_run_schedule(PreStartup);
+        let _ = world.try_run_schedule(Startup);
+        let _ = world.try_run_schedule(PostStartup);
+
+        Self::run_main(world);
```

Comment:
Hmm. I don't think this is the right approach here. If we want to have some "do a clean pass" logic, we should be pulling that out into the system that actually calls this method and changing the code there.

Reply:
Hm, this function is not called from a system, but rather directly by `app.run()` directly. Shall I just move the `run_main()` call there?

Reply:
Yes please, I think that's the way.